elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/timegrains/Day.java

```diff
@@ -34,10 +37,13 @@ public Day deserialize(Object val) {
 
             try {
                 if (val instanceof String) {
-                    date = new Day(new Timestamp(FORMATTER.parse((String) val).getTime()));
-                } else {
-                    date = new Day(FORMATTER.parse(FORMATTER.format(val)));
+                    try {
+                        date = new Day(new Timestamp(FORMATTER.parse((String) val).getTime()));
+                    } catch (ParseException pe) {
+                        date = new Day(new Timestamp(ISO_FORMATTER.parse((String) val).getTime()));
+                    }
                 }
+                date = new Day(FORMATTER.parse(FORMATTER.format(val)));
```

Comment:
This should be inside an else block otherwise it will get executed even when val is instanceof string